### DISCUSSION TITLE
Update Config for Custom Span Processor at Runtime

### DISCUSSION THREAD

**Discussion by codeforaliving** [open-telemetry/opentelemetry-java-instrumentation#6800] [Q&A]
We are using 1.14 of the instrumentation library.  We have a custom SpanProcessor within an agent extension jar.  What is the recommended way to update configuration values in the extension during runtime from the main application?   Thanks!

**Comment by mateuszrzeszutek** [comment]
Hey @codeforaliving ,
One way to do that would be to keep the state in the bootstrap class loader (have a separate class there just for storing whatever info you need); classes in the bootstrap CL are accessible both in the agent space and in the application space. One drawback of this is that we don't currently support this use case in extensions; you'd have to make a [distro](https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/distro) instead, or add classes to bootstrap class loader manually, which usually involves hacking around things.
Another would be to do something similar to https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/6191, where you could just add the necessary info to the context -- you'd need a bridge instrumentation that translates between agent and application OpenTelemetry API calls though.

**Reply by laurit** [reply]
I believe the simplest way would be to use [System.setProperty](https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#setProperty(java.lang.String,%20java.lang.String)) and [System.getProperty](https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#getProperty(java.lang.String))
